pkg/templates/openapi/default.go

```diff
@@ -42,41 +44,80 @@ func GenSchemaDefaultPatch(ctx context.Context, schema *openapi3.Schema) ([]byte
 					group: name,
 				}
 			case len(nodes) == 1:
-				np := nodes[0].GetPatch()
-				if len(np) == 0 {
+				def := nodes[0].GetDefault()
+				if def == nil {
 					resultChan <- patchResult{
 						group: name,
 					}
 
 					return
 				}
 
+				// Generate patch.
+				var b []byte
+
+				p, err := sjson.SetBytes(b, nodes[0].GetID(), def)
+				if err != nil {
+					resultChan <- patchResult{
+						group: name,
+						err:   fmt.Errorf("set patch node error for %s %s: %w", nodes[0].GetID(), string(p), err),
+					}
+				}
+
 				resultChan <- patchResult{
 					group: name,
-					patch: nodes[0].GetPatch(),
+					patch: p,
 				}
 			case len(nodes) > 1:
-				var (
-					p   = nodes[0].GetPatch()
-					err error
-				)
-
-				for i := len(nodes) - 1; i >= 0; i-- {
-					np := nodes[i].GetPatch()
-					if len(np) == 0 {
-						continue
+				if nodes[0].GetDefault() == nil {
+					resultChan <- patchResult{
+						group: name,
+						patch: nil,
 					}
 
-					if len(p) == 0 {
-						p = np
+					return
+				}
+
+				// Generate root patch.
+				var b []byte
+
+				p, err := sjson.SetBytes(b, nodes[0].GetID(), nodes[0].GetDefault())
+				if err != nil {
+					resultChan <- patchResult{
+						group: name,
+						err:   fmt.Errorf("set patch node error for %s %s: %w", nodes[0].GetID(), string(p), err),
+					}
+				}
+
+				for i := 0; i < len(nodes); i++ {
+					np := nodes[i]
+					if np.GetDefault() == nil {
 						continue
 					}
 
-					p, err = jsonpatch.MergePatch(p, np)
+					// Generate sub node patch.
+					var nb []byte
+
+					npb, err := sjson.SetBytes(nb, nodes[i].GetID(), nodes[i].GetDefault())
 					if err != nil {
 						resultChan <- patchResult{
 							group: name,
-							err:   fmt.Errorf("merge patch error for %s %s: %w", nodes[i].GetID(), string(p), err),
+							err:   fmt.Errorf("set patch node error for %s %s: %w", nodes[0].GetID(), string(p), err),
+						}
+					}
+
+					var (
+						pid = np.GetParentID()
+						pr  = gjson.ParseBytes(p)
+					)
+
+					if pe := pr.Get(pid); pe.Exists() {
+						p, err = jsonpatch.MergePatch(npb, p)
```

Comment:
if an error occurs, the p will be reset, we should allocate a new local var to receive the merged result, and then assign it to p if no error occurs.

Reply:
Update to return error while error occurs